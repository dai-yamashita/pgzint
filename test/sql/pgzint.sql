CREATE EXTENSION pgzint;

SELECT bc_pdf417('test');
SELECT bc_qrcode('test');
SELECT bc_code128('test');

SELECT getzintsymbolid('BARCODE_PDF417');
SELECT getzintsymbolconstant(55);
