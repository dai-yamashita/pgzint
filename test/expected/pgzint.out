CREATE EXTENSION pgzint;
SELECT bc_pdf417('test');
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              bc_pdf417                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 \x89504e470d0a1a0a0000000d494844520000026a0000007e08020000001b68aa8c0000035849444154789ceddd416ee3381440c1f1c0f7bf727adb180283ffd894e44eaa96814252348d07aff8fae7c37c7d7dfde72fafd7ebb66726d6714eb9f3dd4f99ece164f6cfff4cf746de73eabd2623af9eddc3bd773f75c23fff3b78e749b8ce9ddfe55373adfff56f5c1b00209f00d0c9270064f20900997c0240269f0090c9270064f20900997c0240269f0090c9270064f20900997c0240269f0090c9270064f20900997c0240f67e7a01dc61ef2ef5bd714e99cc35b9477e6fe4d5dec87bf7da7f8fd9f75cf716cffaaeeff593f9f50900997c0240269f0090c9270064f20900997c0240269f0090c9270064f20900997c0240269f0090c9270064f20900997c0240269f0090c9270064f20900d9fbe905fc1493bbe64fb96ee453aedb8d75e4532623dfb9f393f5dc79ea26d6d9affbbc56cf9ebac95c7bebb9730ff99d5f9f0090c9270064f20900997c0240269f0090c9270064f20900997c0240269f0090c9270064f20900997c0240269f0090c9270064f20900997c0240269f0090bd9f5ec0f734b9d77ef2cc6af2cce4f6f9bd7bed4fcd3e716a9cc9ceef7d16ab536bde1bf9ce373de5bad93fed3bf8693bcf9ff3eb130032f904804c3e0120934f00c8e4130032f904804c3e0120934f00c8e4130032f904804c3e0120934f00c8e4130032f904804c3e0120934f00c8e41300b2f7d30be04fadb7d8effdd7ebf53a32f29e53eb59ff6bef9989c938a7f6706fcd7b2b3c35d77527736f3d9367f6cee1facca9b371ea2db8825f9f0090c9270064f20900997c0240269f0090c9270064f20900997c0240269f0090c9270064f20900997c0240269f0090c9270064f20900997c0240269f0090bd9f5ec0f734b9237ee2d4bdf693913fcdde0a273b76ea9989c9c813a7d673cae42d262b9c8cf3f96775cfe46cecbdfba953c7fff3eb130032f904804c3e0120934f00c8e4130032f904804c3e0120934f00c8e4130032f904804c3e0120934f00c8e4130032f904804c3e0120934f00c8e41300b2f7d30bf829d6dbdeefbc117e32fb6af2cc64aec9c87fe3fe5cb79ed5de7af6f67932f26aefb4fc8d9ff2a9efc53acea991ef3c993f995f9f0090c9270064f20900997c0240269f0090c9270064f20900997c0240269f0090c9270064f20900997c0240269f0090c9270064f20900997c0240269f0090fd028c2941ff47af46390000000049454e44ae426082
(1 row)

SELECT bc_qrcode('test');
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    bc_qrcode                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 \x89504e470d0a1a0a0000000d4948445200000124000001240802000000099bdbbb00000cee49444154789cedddd18ee4360c44d174e0ffffe5cdab0d840809569514e09eb76cd4b2dd3384cda144fffe32f8f3e7cfbffefbeff7fbcff1ef31d37954e7508dafceb333cff41c3adf4367cee9fcaae36e7e76539beb9afe7c37fe96cf08e05f116c4008c106847c1e4c3bcfe8e544c367fde99c6f8edcc971ceaa73d8e4158e7caf736e9d31eef357e5a2aab8e0ce0684106c4008c1068494399baa6ef33998b9a631cde536d7588d51d9d4f7dc35b4caa9bcd741956792b30107106c4008c106843ca70eac5a53578daf9e9ba739cfa9b59a9d793a9fdde43fd3f18e9aa72a4f4ee6db15ee6c4008c10684106c40883d679be6249d1cc35153eaccbfa9076eea7e9b7cc3b15770ea541db5f3d9644d8f3b1b1042b00121041b1052e66caae7d74dbf8769fed639ee86636fd5664c755cd5b53bd64f76e69fe6518e3d728ecf72670342083620846003423e399bbbc6e2eeef97ec1f98ac07baeb87aa9e1c8eefd371bd9de3566336b8b30121041b1042b001214f723f8f3b6f71acc3ac38f66855f3bbf39f64ffc929773db6735c15ee6c4008c10684106c40c84fd5f3c39d876cd6049e7a1fd7d4a99e1f2aee35999b9a98fbf7b6833b1b1042b00121041b10d27a3f9be339b89ac7bdf7ec863e84eefd6c15c7bb103ac77a4beeaf53d51255df2d77362084600342083620e4513da3abfaefb97b6ca8fa346e6a41eedcac33cff47ca673befddf73bfcd3cef31dcd98010820d0821d88090726de4679068fd6167feb74dad2c59cb72ec7d72af1dadc65454eb03a73f97e43ac98aea58dcd98010820d0821d88090f1dac81b721e554e589d837b5de2f4b32a9b9cd67dfec93c5995834de7e1ce0684106c4008c10684b47a90b8eb128e1ef4eef39fce5371e45153aaf3ec7c764af5f782e9fc9df11dd4d9800308362084600342646b233b6ece9d361c79a9aac789ea1d03d3319df1c9791c7f9ba8d08304388c600342083620a47c50dee4069df1ee63b99daaa1b9f783397ea68edc3ef92e07d5ef2777362084600342083620a47ca776f2bd67eefefbd339a7e7a3eabbe8a8176dbe7fc7cf7ab32f6edae7737a0ed59caa3ea2dcd98010820d0821d88090e7fd1f9bda8ba3a7bcc3260f719cf3b44eb5c96f55f549f7bec1e99cc9fe2e9bbf5f7067034208362084600342c66b23933d181deb036fd823e75eaf58cdd93907c7671d6b2fa7e7b339b7cdde39ea6cc001041b1042b001213fd5b3697900c3fea2ea1c367bab923952b2576135cfcdbd1c3bc79a8ee9ccffa6fa9bc51b77362084600342083620a4ccd954dcfd124ff54874d46192f5b70ec7bac76a4cc7cd7d23e941025c84600342083620e471ef479af69650f5a8e83cc7abf2a58a7beda5fb9c4fe596ee1cafe2ae1372670342083620846003429e64bd4865d31fc5b126f34d957faad6f8a9f66e758eb5e94fd3e1c8e5aa791c6b3eb9b30121041b1042b001219fbe9137f4fddbd4433ae33bf5c064ff954d0db09ab39a67730e9d636d6cf249f7df08de36bfe7dcd98010820d0821d88090cf43f9a6e6d6715b8f78c79eab37475f44f79e4007554d52f53d38d0eb1fb808c10684106c4048f94eed0e47bff8ce78559d6a7aedd3eb4de60c9bfcc7ddd35255af73d77537d74b0f12e022041b1042b001214ff53f1c7d023be31d7d3e92ef30a86cea48aabd6a9dcf5654fb151d7bfc1cfb151d393f77362084600342083620e4e9e42d8edca9c35dff49f608e9cca37a97c0f4b8eef1a7f6cbb96bc86f9d9f0577362084600342083620e4e778365571bc2f2bb90fcad15bd2dddfffd4b53b7e2ea7f25ef6b30187116c4008c106847cd646aafa61b8dfb7d6193fa57a9fc1cdbd252baa7d62a78e7b43ff18fa46021721d88010820d08693dd43aea158e7ad1e6b8538eeb9d8e77ece5eb70af4d75cc33a5aa85bec770670342083620846003427e8ebcab7560733d4af59cadea8de9e89d5851e5879b73d8ccdf3996aadebb39870a7d2381c308362084600342cabe9115d55ea94e5eb1c9c11cef2873f4cc48f6a99fcea3eaabb919d3a1aa036fae9d1e24c045083620846003423e0f9aee3e8aeebe8837d4c4547df655f534558eb7c9eb1c3961e73c37e7e3d8c7c89d0d0821d88010820d087954b585ce671dfde5a775b9e971937d323be7b0a937ba6b68a7de9f56d95c8ba33f0a7736208460034208362064d5ebdfb16fcab1df6c7aac0e473ffae93cee1e248e3d63c9badc0dfd72c8d9800308362084600342cab59137ac219c72f75474f7eaa838d6e955f34fb9fb79def61e82cd18ee6c4008c10684106c4048abcee6d807d519ffe6e80fe9a6dabfe7f819396a50b7ed4febcca9ca7bab737be3ce0684106c4008c10684947d23a7cfacaae7e6ea597cda87d0dd435f958b6e72a79b7bba4c6dd6824eeb60aafd6953dcd98010820d0821d88090f241d9b15e51752c77ed6533e7664de9f4b8ee7583ee7992ef4e70f4bdac8e5be1ce0684106c4008c106843c8e5a8a3b07531d6b5ab749aec74bf60871d4541d35c0cd77b2d1397f7a90001721d88010820d0879a6398ca337c3d40deb1e937d41a6eb2a2b8e1a6067fecef9a8d62baad6943ad65572670342083620846003425aef67ebe8e4156fc99a8faa96a8cacd54f9db660d61b23e39fd1d70d4caa6c79deae481dcd98010820d0821d88090b26fa4aa9e561ed8b0ce4d552f7a53d5824eed85ebccdff9ac639e0ef777323d6e077d2381c308362084600342cabe9195693dad32ddd3357de676ecd3db5cfb66ede5669ecef9bca97215f7fad8ce39248f450f12e022041b1042b0012165cee6786f5867cca6d77f87aa5f7c45d5d35f554f73af9fece4d8c97efad5b99dfa7d7ee3ce0684106c4008c10684b4ea6caa7e18ee5a90aa96d5b1e953a25a27d919bfe9d3d899a7736e8efd699bfcd0d1dfa542df48e000820d0821d880904fcee65e8bd8f9acaa5f856a8f5965337eb3feb3d3cfb01a3f3db76accf4dc3a54ef0350d5e8547b1ddfb8b30121041b1042b001219f87cb64bf3e551da3a2ea5de1e0ee7de278afc0749ee9f94cd7799eda2f57e99c0377362084600342083620a4f550bba9c338de21d619df31aded4cbf07f7b5579f551d4bb53f6d43759e9dcf56633ae7d3f92c77362084600342083620a45567533d1f7f0e2caa93386a448e3a95bb8ee4585b985c6b7a5bbffecd712bdcd98010820d0821d8809055dfc8ce7847af8b6a7c87a3c6e2a8b955f34ffb2eaa7263771daf1aa3e2eeb149af7fe022041b1042b001214fe7f9fe546fc3e998ea1ca63964f2bd619be3baf39f53fbd32aaa6b54fd0e5431525d2377362084600342083620e471f79ce83cd76eb8f7d14d8fdb712ad79d72d7c154fbe554bd463b6336e7c99d0d0821d88010820d08f9ac8d4cee4f733cf7bf6d6a838e779a6dc66ff284e9754dafddd1975fb5b7f0cdb1feb3f359fa460207106c4008c106843c9be7fece98e9fab14ded48f5bc3efdf7e977e8a0cab5a6c7ea70e4f0d36b9c9e8f2a6f676d247000c10684106c4048f94eed8ecd737387e3b979730ed59caa3e2b9d1c66d3bf5155e79cd6e83af3bca97e0f1dbf9f9b73e6ce0684106c4008c1068458fa465692fd27abe33aaeab33dedd97bff3ef954e5eeaa8c776e6d9987eb79b9e289dcf72670342083620846003425a3d481c7bc33adc7b931c7d262a9b7a9a2adfdbec6dabb873b9cd357638fad354f37367034208362084600342ca5eff1daa3e0dd598ceb1dcf961c5d1f3d0ddf3a3c3f1339dce331de3c8b12bac8d04fe0708362084600342c60ffac9defaaafade664cc5917f7638faaf7424eb51ee7718a8d6af76d037123880600342083620a4ec1be95877578daf74fa4f56e3a773aac65754f5ab69ff0fd5cfb7f3d98a7b7f5a35663a6765b3cf90b591c001041b1042b00121e503a86adf54eb24ccebfd3a547bae3af3ab3e3b9da7337ff2daab793635b71bd68b56b8b30121041b1042b00121bf4d3ea0ea4fe89eb39aff546ee6ce871d9fade6d9e454d33153c9e3b29f0db808c10684106c40c8d37966ddf4c6a84cd7f839e6ef8c9f9aae51dc48e61bd3f1b7e5a29d7936f97fe777953b1b1042b00121041b1052ee67ab4cf7a7a972c2e9b94d6b2cc9358a8e359f8ee34eaf3759f3accea1f359559fcfcebfbf71670342083620846003425aef679b4aee3b72afd3ab24fb616c7a60a8e6dcf49571bf43a263f3b3d8f459a16f247000c10684106c40c8f3fe0f770ffa4d4d667aaccef86a8ca3cfca0dbdf25535b14a725de874bdee66fe0dd6460207106c4008c106843cd5ff70f40ba93eeb782f5675acb7e9b3beaa26b9c96d1cb984637f97fb1d6b9d73981ecb51177de3ce0684106c4008c1068494399b8aea3958b5df49f59e37d53938ae6b4ab59652957bdff00e09c777c29d0d0821d88010820d08b1e76c6faa1a9a63bfd93497dbac8d3cd5fbdeb1cfadfaac7bcecdfe3a37fa460287116c4008c1068494399b234fa8fedd5d57999e9b6abf9caaa6e4c8733ae337f33bbeff0ec7bb2b3a9fed5c2f77362084600342083620e493b339f61775a87aacbb7b87a8f281e9bb0736e3dd7bf0de4ef58051ada775f7ffe4ce0684106c4008c10684fc0352ef6d02b14218890000000049454e44ae426082
(1 row)

SELECT bc_code128('test');
                                                                                                                                                                                                                                                                                                                                                                    bc_code128                                                                                                                                                                                                                                                                                                                                                                    
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 \x89504e470d0a1a0a0000000d494844520000009e000000640802000000f8ee1e530000012e49444154789cedd1410ac320144041edfdef6c178522d886d0ee8637ab10f4abbc39366bad31c69cf3fdfd72fed97d5b7fba337fdffbdb1dae4fbfbef3ffafb87feef584f3a5e7fc73cebee6f1f136019496555a566959a5659596555a566959a5659596555a566959a5659596555a566959a5659596555a566959a5659596555a566959a5659596555a566959a5659596555a566959a5659596555a566959a5659596555a566959a5659596555a566959a5659596555a566959a5659596555a566959a5659596555a566959a5659596555a566959a5659596555a566959a5659596555a566959a5659596555a566959a5659596555a566959a5659596555a566959a5659596555a566959a5659596555a566959a5659596555a566959a5659596555a566959a5659596555ad61305823fc8e0b941bd0000000049454e44ae426082
(1 row)

SELECT getzintsymbolid('BARCODE_PDF417');
 getzintsymbolid 
-----------------
              55
(1 row)

SELECT getzintsymbolconstant(55);
 getzintsymbolconstant 
-----------------------
 BARCODE_PDF417
(1 row)

